Address SaferCpp issues in WebPopupMenu.cpp

kmonsen opened the pull request:
#### ef7a6271a65f19a6570c7471775248431ff478f4
<pre>
Address SaferCpp issues in WebPopupMenu.cpp
<a href="https://bugs.webkit.org/show_bug.cgi?id=291017">https://bugs.webkit.org/show_bug.cgi?id=291017</a>
<a href="https://rdar.apple.com/problem/148541490">rdar://problem/148541490</a>

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
(WebKit::WebPopupMenu::didChangeSelectedIndex):
(WebKit::WebPopupMenu::setTextForIndex):
(WebKit::WebPopupMenu::populateItems):
(WebKit::WebPopupMenu::show):
(WebKit::WebPopupMenu::hide):

Canonical link: <a href="https://commits.webkit.org/293386@main">https://commits.webkit.org/293386@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/1fd744abffcccbaf5748dacf9f7a37cbc67f4080

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/98728 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/18361 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/8598 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/103853 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/49318 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/18654 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/26812 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/75163 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/32310 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/101732 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/14169 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/89162 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/55520 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/13955 "Passed tests") | | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/48698 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/83904 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/7205 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/106224 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/25818 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/18825 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/84135 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/26195 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/85363 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/83622 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/21128 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/28273 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/5948 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/19537 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/25776 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/30958 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/25594 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/28914 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/27169 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
```diff
@@ -60,66 +60,67 @@ void WebPopupMenu::disconnectClient()
-    m_popupClient->popupDidHide();
-    if (newIndex >= 0)
-        m_popupClient->valueChanged(newIndex);
+    if (CheckedPtr popupClient = m_popupClient) {
```
We prefer early returns in WebKit in general. We sometimes make exceptions for one liners but this looks like something that should have kept the old style.

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
```diff
@@ -60,66 +60,67 @@ void WebPopupMenu::disconnectClient()
-        return;
-
-    m_popupClient->setTextFromItem(index);
+    if (CheckedPtr popupClient = m_popupClient)
```
With one line like this seems fine though.

cdumez commented on Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
```diff
@@ -60,66 +60,67 @@ void WebPopupMenu::disconnectClient()
-    m_page->setActivePopupMenu(nullptr);
-    m_popupClient->popupDidHide();
+    if (RefPtr page = m_page.get()) {
+        if (CheckedPtr popupClient = m_popupClient) {
```
Please use early returns. This is exactly what we're trying to avoid in WebKit.

cdumez reviewed (approved):
LGTM

kmonsen committed:
Address SaferCpp issues in WebPopupMenu.cpp
https://bugs.webkit.org/show_bug.cgi?id=291017
rdar://problem/148541490

Reviewed by Chris Dumez.

* Source/WebKit/SaferCPPExpectations/UncheckedCallArgsCheckerExpectations:
* Source/WebKit/SaferCPPExpectations/UncountedCallArgsCheckerExpectations:
* Source/WebKit/WebProcess/WebCoreSupport/WebPopupMenu.cpp:
(WebKit::WebPopupMenu::didChangeSelectedIndex):
(WebKit::WebPopupMenu::setTextForIndex):
(WebKit::WebPopupMenu::populateItems):
(WebKit::WebPopupMenu::show):
(WebKit::WebPopupMenu::hide):

Canonical link: https://commits.webkit.org/293386@main
